### PR TITLE
[PB-2228]: feat/real time changes apn

### PR DESCRIPTION
**Pull request by jzunigax2**
Related issue:
https://inxt.atlassian.net/browse/PB-2228

Implemented APN to send push notifications to mac clients signaling changes in their virtual storage system. In every place where a post to notifications was triggered a method attempts to find any macos notification tokens (table created in internxt/drive-server-wip#341) and if there are any it attempts to send a notification.

### CODE REVIEW

File: src/config/initializers/apn.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+export default class Apn {
+  private static instance: Apn;
+  private client: http2.ClientHttp2Session;
+  private readonly maxReconnectAttempts = 3;
+  private reconnectAttempts = 0;
+  private reconnectDelay = 1000;
+  private readonly bundleId = process.env.APN_BUNDLE_ID;
+
+  private readonly apnUrl =
+    process.env.NODE_ENV === 'production' ? 'https://api.push.apple.com' : 'https://api.sandbox.push.apple.com';
```
Comment (sg-gs) [inline src/config/initializers/apn.ts:14]: This should be an environment variable, not a ternary with hardcoded URLs, especially if we want to open the option to modify this values remotely without re-deployments

File: src/config/initializers/apn.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    return new Promise((resolve, reject) => {
+      if (!this.client || this.client.closed) {
+        this.connectToAPN();
+      }
+      const headers = {
+        'apns-topic': `${this.bundleId}.pushkit.fileprovider`,
+        authorization: `bearer ${this.generateJwt()}`,
+      };
+
+      const options = {
```
Comment (sg-gs) [inline src/config/initializers/apn.ts:105]: The http Node.js client has [typings](https://nodejs.org/api/http.html) for the options, this hardcoding is not the best way to handle it. Type the options const as [Partial](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) so you get the autocomplete.

File: src/config/initializers/apn.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      req.end();
+
+      let statusCode = 0;
+      let data = '';
+
+      req.on('response', (_, status) => {
+        statusCode = status || 0;
+      });
+
+      req.on('data', (chunk) => {
```
Comment (sg-gs) [inline src/config/initializers/apn.ts:132]: This event-based request usage is not the best way to handle this consistently, why not use `axios` if you are not streaming anything?

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/config/initializers/apn.ts (hunk 3 above)
Comment (jzunigax2) [reply src/config/initializers/apn.ts:132]: Axios doesn't support http/2, and APN only accpets connections over this protocol

**Review by sg-gs — APPROVED** [review]

**Comment by sg-gs** [comment]
Secrets have been set in production correctly. Merging @jzunigax2 @PixoDev

**Pull request merged**